packages/solid-query/src/__tests__/createQuery.test.tsx

```diff
@@ -0,0 +1,6485 @@
+      createQuery(key, queryFn1)
+      createQuery(key, queryFn2)
+      return null
+    }
+
+    render(() => (
+      <QueryClientProvider client={queryClient}>
+        <Page />
+      </QueryClientProvider>
+    ))
+
+    expect(queryCache.find(key())!.options.queryFn).toBe(queryFn1)
+  })
+
+  // TODO(lukemurray): this seems like a react implementation detail
+  it.skip('should render correct states even in case of useEffect triggering delays', async () => {
+    const key = queryKey()
+    const states: CreateQueryResult<string>[] = []
+
+    // @ts-expect-error - we skip this test
+    const originalUseEffect = NotReact.useEffect
+
+    // Try to simulate useEffect timing delay
+    // @ts-ignore
+    NotReact.useEffect = (...args: any[]) => {
+      originalUseEffect(() => {
+        setTimeout(() => {
+          args[0]()
+        }, 10)
+      }, args[1])
+    }
+
+    function Page() {
+      const state = createQuery(key, () => 'data', { staleTime: Infinity })
+      createRenderEffect(() => {
+        states.push({ ...state })
+      })
+      return null
+    }
+
+    render(() => (
+      <QueryClientProvider client={queryClient}>
+        <Page />
+      </QueryClientProvider>
+    ))
+    queryClient.setQueryData(key(), 'data')
+    await sleep(50)
+
+    // @ts-ignore
+    NotReact.useEffect = originalUseEffect
+
+    expect(states.length).toBe(2)
+    expect(states[0]).toMatchObject({ status: 'loading' })
+    expect(states[1]).toMatchObject({ status: 'success' })
+  })
+
+  it('should batch re-renders', async () => {
+    const key = queryKey()
+
+    let renders = 0
+
+    const queryFn = async () => {
+      await sleep(15)
+      return 'data'
+    }
+
+    function Page() {
+      createQuery(key, queryFn)
+      createQuery(key, queryFn)
+      renders++
+      return null
+    }
+
+    render(() => (
+      <QueryClientProvider client={queryClient}>
+        <Page />
+      </QueryClientProvider>
+    ))
+
+    await sleep(20)
+
+    // Since components are rendered once
+    // There wiil only be one pass
+    expect(renders).toBe(1)
+  })
+
+  it('should batch re-renders including hook callbacks', async () => {
+    const key = queryKey()
+
+    let renders = 0
+    let callbackCount = 0
+
+    const queryFn = async () => {
+      await sleep(10)
+      return 'data'
+    }
+
+    function Page() {
+      const [count, setCount] = createSignal(0)
+      createQuery(key, queryFn, {
+        onSuccess: () => {
+          setCount((x) => x + 1)
+        },
+      })
+      createQuery(key, queryFn, {
+        onSuccess: () => {
+          setCount((x) => x + 1)
+        },
+      })
+
+      createEffect(() => {
+        renders++
+        callbackCount = count()
+      })
+
+      return <div>count: {count()}</div>
+    }
+
+    render(() => (
+      <QueryClientProvider client={queryClient}>
+        <Page />
+      </QueryClientProvider>
+    ))
+
+    await waitFor(() => screen.getByText('count: 2'))
+
```

Comment:
the whole expect was removed here: https://github.com/TanStack/query/pull/4211/commits/0224d6b29073d97426503312723af4c2cc50d9ec

I think we should just consistently expect a render count of 3. with no check at all, the test is rather useless:

```suggestion
    await waitFor(() => screen.getByText('count: 2'))

    // Should be 3 instead of 5
    expect(renders).toBe(3)
```